android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -387,12 +387,19 @@ constructor(
         ?: emptyList()
     }
 
-    /** This function combines all string indexes to comma separated */
+    /** This function combines all the string indexes to a comma separated list */
     fun joinToString(source: MutableList<String?>): String {
-      source.removeIf { it == null }
-      val inputString = source.joinToString()
-      val regex = "(?<=^|,)[\\s,]*(\\w[\\w\\s]*)(?=[\\s,]*$|,)".toRegex()
-      return regex.findAll(inputString).joinToString(", ") { it.groupValues[1] }
+      return joinToString(source, DEFAULT_REGEX, DEFAULT_STRING_SEPARATOR)
+    }
+
+    /**
+     * This function combines all string indexes to a list separated by the separator and regex
+     * defined by the content author
+     */
+    fun joinToString(sourceString: MutableList<String?>, regex: String, separator: String): String {
```

Comment:
We we use the default regex and separator constants as default parameters on this function. I think kotlin functions using default params are transformed into overloaded functions in the JVM at runtime and should work well with JEXL, maybe not.